api_server.py

```
@@ -74,38 +74,56 @@ def format_hours(minutes: Optional[int]) -> Optional[str]:
     return f"{minutes / 60.0:.2f}h"
 
 
+def format_availability_data(now: datetime, end_time: Optional[Any]) -> Dict:
```

Ruff (0.15.0)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `end_time`

(ANN401)